src/Coordination/KeeperStateMachine.cpp

```diff
@@ -308,7 +342,11 @@ int KeeperStateMachine::read_logical_snp_obj(
                             s.get_last_log_idx(), latest_snapshot_meta->get_last_log_idx());
             return -1;
         }
-        data_out = nuraft::buffer::clone(*latest_snapshot_buf);
+        if (bufferFromFile(log, latest_snapshot_path, data_out))
```

Comment:
Print errno?

Reply:
errno has been printed in the inner function.